### PR TITLE
Add support for variables in Custom JS variable

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Added code to support variables in Custom JS variable
Fixes: #PG-3109, #212

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: Context/BaseContext.php [hunk 1]
```diff
@@ -160,6 +160,23 @@ protected function generatePublicContainer($container, $release)
         return $containerJs;
     }
 
+    public function getPreConfiguredVariablesJSCodeResponse($context)
```
Comment (michalkleiner) [inline Context/BaseContext.php:163]: Preconfigured would be one word here.
```suggestion
    public function getPreconfiguredVariablesJSCodeResponse($context)
```

Comment (michalkleiner) [reply Context/BaseContext.php:163]: Same with all other occurrences.

Comment (michalkleiner) [reply Context/BaseContext.php:163]: But I see it's already an established pattern (despite being incorrect) even at the folder name level, so all good.

**Comment by AltamashShaikh** [comment]
for cases like `var {{foo}}={{bar}}` use `this[{{foo}}] = {{bar}};`

**Review by snake14 — APPROVED** [review]
Looks good :+1: 
I was able to output the results of a CustomJS variable which referenced another CustomJS variable without any issues.

**Pull request merged**